### PR TITLE
Add NetworkAvailabilityChanged event to INetworkManager

### PR DESCRIPTION
**Pull request by stone-w4tch3r**
added network status changed event handler to INetworkManager

### CODE REVIEW

File: src/Arbus.Network/Implementations/WindowsNetworkManager.cs [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 
 namespace Arbus.Network.Implementations;
 
 public class WindowsNetworkManager : INetworkManager
 {
+    public NetworkAvailabilityChangedEventHandler? NetworkAvailabilityChangedEventHandler { get; set; }
+
+    public WindowsNetworkManager()
+    {
+        NetworkChange.NetworkAvailabilityChanged += NetworkAvailabilityChangedEventHandler;
```
Comment (evgenyvalavin) [inline src/Arbus.Network/Implementations/WindowsNetworkManager.cs]: I'm not sure the use of NetworkChange class is what we need. NetworkChange class "Allows applications to receive notification when the Internet Protocol (IP) address of a network interface, also called a network card or adapter, changes."
What we actually need is to rise an event when IsNetworkAvailable property value changes.
How about simple:

`public event EventHandler<bool> NetworkAvailabilityChanged;` ?

File: src/Arbus.Network/Implementations/WindowsNetworkManager.cs [hunk 2]
```diff
@@ -1,10 +1,18 @@
 ﻿using Arbus.Network.Abstractions;
 using System.Net.NetworkInformation;
+using static Arbus.Network.Abstractions.INetworkManager;
 
 namespace Arbus.Network.Implementations;
 
 public class WindowsNetworkManager : INetworkManager
 {
+    [Obsolete("Not implemented yet", error: true)]
```
Comment (stone-w4tch3r) [inline src/Arbus.Network/Implementations/WindowsNetworkManager.cs]: Unfortunately, c# doesn't have [NotImplementedAttribute] or something similar. WA - [Obsolete]

Comment (stone-w4tch3r) [reply src/Arbus.Network/Implementations/WindowsNetworkManager.cs]: @evgenyvalavin i don't see your comment in PR. May be your "PR review" was not "finished"? You can check it in top-right corner on files page

**Review by evgenyvalavin — APPROVED** [review]
LGTM

**Pull request merged**